dev/core/src/com/google/gwt/dev/jjs/ast/JCaseStatement.java

```diff
@@ -17,28 +17,68 @@
 
 import com.google.gwt.dev.jjs.SourceInfo;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Java case statement.
  */
 public class JCaseStatement extends JStatement {
 
-  private JExpression expr;
+  private List<JExpression> exprs;
 
   public JCaseStatement(SourceInfo info, JExpression expr) {
     super(info);
-    this.expr = expr;
+    this.exprs = Collections.singletonList(expr);
+  }
+
+  public JCaseStatement(SourceInfo info, Collection<JExpression> exprs) {
+    super(info);
+    this.exprs = Collections.unmodifiableList(new ArrayList<>(exprs));
```

Comment:
We are still compiling with `--release 8`, so that's a no-go without more changers to the build files, probably best done separately (and then start chasing all of the other places that should be cleaned up in this way if we think it matters). 

This is in an effort to keep gwt-servlet etc compatible on the server with Java 8.

Reply:
Hm ok. I only had the release notes of 2.11 in mind which states that it is likely the last minor release that supports running on Java 8.